src/main/java/org/eln2/mc/common/cells/foundation/objects/SimulationObjectSet.kt

```diff
@@ -0,0 +1,62 @@
+package org.eln2.mc.common.cells.foundation.objects
+
+/**
+ * This represents an immutable set of simulation objects.
+ * All possible simulation objects are stored in fields, because there won't be many (probably only electrical, thermal and mechanical),
+ * so there is no reason not to skip an allocation for a map/array.
+ * */
+class SimulationObjectSet(objects: List<ISimulationObject>) {
+    constructor(vararg objects: ISimulationObject) : this(objects.asList())
+
+    private var electrical: ElectricalObject? = null
```

Comment:
I'm wondering if this should be a list of simulator objects instead of being directly made here.